de.lucaswerkmeister.corebird.json

```diff
@@ -0,0 +1,65 @@
+{
+  "app-id": "de.lucaswerkmeister.corebird",
+  "runtime": "org.gnome.Platform",
+  "runtime-version": "3.26",
+  "sdk" : "org.gnome.Sdk",
+  "command" : "corebird",
+  "rename-icon" : "corebird",
+  "copy-icon" : true,
+  "finish-args" : [
+    "--share=ipc",
+    "--share=network",
+    "--socket=x11",
+    "--socket=wayland",
+    "--socket=pulseaudio",
+    "--filesystem=host",
```

Comment:
Yeah, it would be nice to lock this down a bit more. It looks like Corebird uses `~/.corebird` for most its files if that exists, otherwise `$XDG_CONFIG_DIR/corebird`. (Presumably, `~/.corebird` came first and is still supported for backwards compatibility, but `$XDG_CONFIG_DIR/corebird` is preferred now.) But when stracing the app, I also see it trying to access a lot of other paths (`~/.libvisual`, `~/.frei0r-1`, `~/.ladspa`, `~/.lv2`, `~/.xkb`, etc.), most of which I haven’t even heard of. I’m not sure how to debug this properly without breaking the app for some users :/

Reply:
If it's using `$XDG_CONFIG_DIR/corebird` I would give it a try with no `--filesystem` switch/setting at all. Then do `journalctl -f` on another terminal and see how much pain you are causing to `Core Bird`, if it actually starts correctly. Maybe it's only trying to read on those `~/.libvisual`, `~/.frei0r-1`, `~/.ladspa`, `~/.lv2`, `~/.xkb` – which I think it's allowed by default, even though you don't specify that (but I'm not sure about that).

Reply:
I’ve pushed a commit which reduces the file system access substantially. Unfortunately, it hard-codes the `~/.config` path without respecting `$XDG_CONFIG_HOME` – any idea how I could fix that?

Reply:
Patch the app to use `g_get_user_config_dir()`. I am very surprised it didn't already.

Reply:
I’m sorry, perhaps I wasn’t clear. The app already uses that ([here](https://github.com/lucaswerkmeister/corebird/blob/1.7.4.2/src/util/Dirs.vala#L32)) – but the manifest doesn’t.

Removing all filesystem permissions breaks the app – it doesn’t use the GNOME theme and can’t save its preferences.

Reply:
Well the theme problem is probably related to your DE. It probably doesn't support XSettings. That is its own discussion.

The Corebird settings problem doesn't make any sense based on that code and is absolutely a bug in the application.

Reply:
The DE is GNOME, yes.

Reply:
On GNOME no settings (like themes) should be coming from disk access. They are over XSettings or DConf (on Wayland). But if `~/.config/gtk-3.0/settings.ini` exists I guess something weird could be going on and you should stop using that file.

Reply:
The theme is installed as a system-level package ([`arc-gtk-theme`](https://www.archlinux.org/packages/community/any/arc-gtk-theme/)) in `/usr/share/themes`.

`~/.config/gtk-3.0/settings.ini` exists, yes, with the following content:
```config
[Settings]
gtk-application-prefer-dark-theme=1
```
I don’t think I created it manually – I think GNOME Tweaks is the most likely candidate. So I’m not sure what you expect me to do when you tell me to “stop using that file”, to be honest – I promise I didn’t set out trying to create the weirdest possible system just to confuse you :)

Reply:
> I think GNOME Tweaks is the most likely candidate.

Select `Arc-dark` or whatever variant in GNOME-Tweaks.

Reply:
After removing the config file and then selecting a different theme and then Arc-dark again in GNOME Tweaks, the file now has `gtk-application-prefer-dark-theme=0` instead of `=…1`. Doesn’t make a difference to Corebird – with `home` read access, it reads the theme correctly (regardless of `settings.ini` content – even if I remove the file, in fact), otherwise it doesn’t.

I don’t mind debugging this behavior further, but I hope it’s not going to block the submission of this app (once other issues are ironed out, such as the rename)? The old manifest had full host access and was still allowed on Flathub, so tho restricted access in this manifest is already an improvement over that.